packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts

```diff
@@ -97,10 +106,90 @@ export class ContextMenu extends _ModuleSupport.BaseModuleInstance implements _M
         this.cleanup.register(this.ctx.eventsHub.on('context-menu:complete', (e) => this.onContext(e)));
     }
 
-    private expandItemsOptions(showing: AgContextMenuItemShowOn): ContextMenuItem[] {
+    private makeGetItemsParams(event: ContextMenuEvent): AgContextMenuGetItemsParams {
+        const { showOn } = event;
+        const { context } = this.ctx.chartService; // TODO: callWithContext
+        switch (showOn) {
+            case 'always':
+            case 'series-area':
+                return { showOn, context };
+
+            case 'series-node': {
+                if (this.pickedNode == null) throw new Error(`this.pickedNode is null`);
+                type Omissions = 'showOn' | 'seriesId' | 'type' | 'event' | 'context';
+                type PickedNode = Omit<AgContextMenuGetItemsParamsSeriesNode, Omissions>;
+                const {
+                    datum,
+                    angleKey,
+                    calloutLabelKey,
+                    colorKey,
+                    labelKey,
+                    radiusKey,
+                    sectorLabelKey,
+                    sizeKey,
+                    xKey,
+                    yKey,
+                }: PickedNode = this.pickedNode;
+                const params: AgContextMenuGetItemsParamsSeriesNode = {
+                    showOn,
+                    context,
+                    seriesId: this.pickedNode.series.id,
+                    datum,
+                    angleKey,
+                    calloutLabelKey,
+                    colorKey,
+                    labelKey,
+                    radiusKey,
+                    sectorLabelKey,
+                    sizeKey,
+                    xKey,
+                    yKey,
+                };
+                for (const k of [
+                    'angleKey',
+                    'calloutLabelKey',
+                    'colorKey',
+                    'labelKey',
+                    'radiusKey',
+                    'sectorLabelKey',
+                    'sizeKey',
+                    'xKey',
+                    'yKey',
+                ] as const) {
+                    if (params[k] === undefined) {
+                        delete params[k];
+                    }
+                }
+                return params;
```

Comment:
Nit: Maybe a bit easier to follow going the other way around (and less verbose, if possible):
```suggestion
                const { datum }: PickedNode = this.pickedNode;
                const params: AgContextMenuGetItemsParamsSeriesNode = {
                    showOn,
                    context,
                    seriesId: this.pickedNode.series.id,
                    datum,
                };
                for (const k of [
                    'angleKey',
                    'calloutLabelKey',
                    'colorKey',
                    'labelKey',
                    'radiusKey',
                    'sectorLabelKey',
                    'sizeKey',
                    'xKey',
                    'yKey',
                ] as const) {
                    if (this.pickedNode[k] !== undefined) {
                        params[k] = this.pickedNode[k];
                    }
                }
                return params;
```